testing/trino-product-tests/src/main/java/io/trino/tests/product/utils/DeltaQueryExecutor.java

```diff
@@ -0,0 +1,139 @@
+                        .connect(jdbcParamsState.url, properties);
+            }
+
+            if (connection == null) {
+                // this should never happen, `javax.sql.DataSource#getConnection()` should not return null
+                throw new IllegalStateException("No connection was created for: " + jdbcParamsState.getName());
+            }
+            if (!jdbcParamsState.prepareStatements.isEmpty()) {
+                try (Statement statement = connection.createStatement()) {
+                    for (String query : jdbcParamsState.prepareStatements) {
+                        try {
+                            statement.execute(query);
+                        }
+                        catch (SQLException e) {
+                            throw new SQLException("Preparatory statement failed: " + query, e);
+                        }
+                    }
+                }
+            }
+            return connection;
+        }
+    }
+
+    private static Driver getDatabaseDriver(String driverClassName)
+    {
+        // TODO Add support for maven coordinate in tempto
+        ArtifactResolver resolver = new ArtifactResolver(USER_LOCAL_REPO, ImmutableList.of(MAVEN_CENTRAL_URI));
+        List<URL> classPath = resolver.resolveArtifacts(getArtifact(driverClassName)).stream()
+                .map(artifact -> {
+                    try {
+                        return artifact.getFile().toURI().toURL();
+                    }
+                    catch (MalformedURLException e) {
+                        throw new RuntimeException(e);
+                    }
+                })
+                .collect(toImmutableList());
+
+        checkArgument(!classPath.isEmpty(), "classPath must not be empty");
+        URLClassLoader classLoader = new URLClassLoader(classPath.toArray(URL[]::new), getPlatformClassLoader());
+        Class<? extends Driver> driverClass;
+        try {
+            driverClass = classLoader.loadClass(driverClassName).asSubclass(Driver.class);
+        }
+        catch (ClassNotFoundException | ClassCastException e) {
+            throw new RuntimeException("Failed to load Driver class: " + driverClassName, e);
+        }
+
+        try {
+            return driverClass.getConstructor().newInstance();
+        }
+        catch (ReflectiveOperationException e) {
+            throw new RuntimeException("Failed to create instance of Driver: " + driverClassName, e);
+        }
+    }
+
+    private static Artifact getArtifact(String driverClass)
+    {
+        if (driverClass.equals("org.apache.hive.jdbc.HiveDriver")) {
+            // Resolve maven artifact at runtime to use different Hive JDBC drivers between Hive and OSS Delta Lake tests because
+            // the old Hive JDBC driver (0.13.1) can't connect to new Delta Lake (2.4.0) and
+            // the new Hive JDBC driver (>= 3.0.0) supports the above Delta Lake version, but it can't connect to old Hive servers (1.2.1).
```

Comment:
`org.apache.hive.jdbc.HiveDriver` is the old or the new driver?

Reply:
Both versions use the same class name. In `DeltaQueryExecutor`, we will use the new version.